### ISSUE TITLE
Raster clims masked array

### ISSUE THREAD

**Issue by vascotenner** [ioam/holoviews#1020]
I would expect that Raster and Image uses the same logic to determine the clims. However, with masked arrays as input, this breaks down:

```
%%opts Raster Image [colorbar=True]
tmp = np.random.rand(10,10)-0.5
tmp = np.ma.masked_where(tmp<=0, tmp)

hv.Image(tmp)+hv.Raster(tmp)
```
![download](https://cloud.githubusercontent.com/assets/3776190/21260008/6935b44a-c384-11e6-8581-534634f3f163.png)

Both images should have same colorscale

**Comment by jlstevens** [comment]
> Both images should have same colorscale

Agreed! Labelling this as a bug.

**Comment by philippjfr** [comment]
Turns out this is a difference in the range calculation, should be fixed by #994.

**Comment by philippjfr** [comment]
Now fixed.
